src/at-unix.c

```diff
@@ -355,9 +467,14 @@ void *at_reader_thread(void *arg)
 
         /* Attempt to read some data. */
         char ch;
+
         int result = read(priv->fd, &ch, 1);
         int why = errno;
 
+        if(handle_globalerrno() != 0)
```

Comment:
It doesn't look nice. It's better to change Nuttx to return -1 from read() and set proper errno. 
In *nix convention system call usually returns -1 if error occur. Then errno can be probed for getting details. http://linux.die.net/man/2/read